share/functions/funced.fish

```diff
@@ -122,8 +130,50 @@ function funced --description 'Edit function definition'
                     continue
                 end
                 echo (_ "Cancelled function editing")
+            else if test -n "$writepath"
+                if not set -q _flag_save
+                    echo (_ "Warning: the file containing this function has not been saved. Changes will be lost when fish is closed.")
+                    set -l prompt (printf (_ 'Save function to %s\? [Y/n]') "$writepath")
+                    read -p "echo $prompt\  " response
+                    if test -z "$response"
+                        or contains $response {Y,y}{E,e,}{S,s,}
+                        set _flag_save 1
+                    else if not contains $response {N,n}{O,o,}
+                        echo "I don't understand '$response', assuming 'Yes'"
+                        set _flag_save 1
+                    end
+                end
+                if set -q _flag_save
+                    # try to write the file back
+                    # cp preserves existing permissions, though it might overwrite the owner
+                    if not cp $tmpname "$writepath" 2>&1
+                        echo (_ "Saving to original location failed; saving to user configuration instead.")
+                        set writepath $__fish_config_dir/functions/(basename "$writepath")
+                        if not cp $tmpname "$writepath"
+                            echo (_ "Saving to user configuration failed. Changes will be lost when fish is closed.")
+                        else
+                            printf (_ "Function saved to %s") "$writepath"
+                            # read it back again - this ensures that the output of `functions --details` is correct
+                            source "$writepath"
+                        end
+                    else
+                        printf (_ "Function saved to %s") "$writepath"
+                        # read it back again - this ensures that the output of `functions --details` is correct
+                        source "$writepath"
+                    end
+                end
             else if set -q _flag_save
                 funcsave $funcname
+            else
+                set -l prompt (_ 'Save function to the fish configuration directory\? [Y/n]')
```

Comment:
To be honest, I'm not a massive fan of this idea.

I'd just suggest running `funcsave $funcname` instead of prompting.

Reply:
(I am happy to be outvoted on this, of course)

Reply:
Let's drop it for now.